### PR TITLE
Bump Version from 2.1.2 to 2.1.3, switch to woofmark plots2 branch

### PR DESCRIPTION
**Pull request by NitinBhasneria**
## Fixes

1. Repoint `woofmark` to pull in fixes from latest https://github.com/jywarren/woofmark/
2. Popup Disappearing problem solved by @NitinBhasneria https://github.com/jywarren/woofmark/pull/3
3. Feature to change heading back into plain text by @shreyaa-sharmaa  https://github.com/jywarren/woofmark/pull/5
4. Refine drag and drop events to prevent interference by @NitinBhasneria  https://github.com/jywarren/woofmark/pull/2
5. Setup jest-puppeteer for UI tests by @Shulammite-Aso https://github.com/jywarren/woofmark/pull/11
6. Added custom insert module by @keshav234156  https://github.com/publiclab/PublicLab.Editor/pull/568
7. Made history icons consistent in demo editor fixes by @Shulammite-Aso  #571
8. Main image box padding and background by @NitinBhasneria  #564 
9. ADDS: Feature to decenter text in wysiwyg and markdown mode by @shreyaa-s #521 
10. Remove file button test added by @NitinBhasneria #574 
11. Adds puppeteer as a dev dependency by @shreyaa-s  #565
12. Add support for class attribute in links by @Shulammite-Aso https://github.com/publiclab/PublicLab.Editor/pull/547
13. Update dependencies in plots2 branch by @Shulammite-Aso https://github.com/jywarren/woofmark/pull/17
14. fix bold texts loosing formatting after converting to markdown and back to wysiwyg by @Shulammite-Aso https://github.com/jywarren/woofmark/pull/15
15. Compact Design for main Image Module by @shreyaa-sharmaa #594

After https://github.com/publiclab/plots2/pull/8285 merges, we can publish this!

### CODE REVIEW

**Comment by jywarren** [comment]
Hi @NitinBhasneria do you think we could try resolving #553 and #475 to try to make reviewing these releases easier, before publishing this one?

**Comment by NitinBhasneria** [comment]
> Hi @NitinBhasneria do you think we could try resolving #553 and #475 to try to make reviewing these releases easier, before publishing this one?

Yea, this would be great.

**Comment by jywarren** [comment]
Ok! I think our checklist for this is now:

* [ ] resolve the issues we're seeing in https://github.com/publiclab/plots2/pull/8135
* [ ] be sure we open a new PR to pull in the latest `jywarren/woofmark` once that's ready

However, I think we can push off the second list item to `v2.1.2` if it slows us down. Thanks! Who is taking up the issues in https://github.com/publiclab/plots2/pull/8135? @NitinBhasneria @shreyaa-sharmaa @keshav234156 @Shulammite-Aso 

Great work, folks!

**Comment by keshav234156** [comment]
@jywarren can we also add #519 in this release?

**Comment by jywarren** [comment]
@keshav234156 i think so; but let's make the top priority fixing the bugs introduced in `v2.1.0` -- thank you!

**Comment by jywarren** [comment]
#519 is now merged into main! So, let's add it to the release notes above. Thanks!

**Comment by jywarren** [comment]
Let's include https://github.com/publiclab/PublicLab.Editor/pull/521 !

**Comment by jywarren** [comment]
https://github.com/publiclab/PublicLab.Editor/pull/564 is also merged now!

**Comment by jywarren** [comment]
Hi @publiclab/editor folks! How are we doing here? We need to prioritize this in order to publish our first release, and won't that feel great!?

Can you check in on the blocking issues related to [my comment above](https://github.com/publiclab/PublicLab.Editor/pull/557#issuecomment-658196723)? Thank you so much! 

cc @sagarpreet-chadha

**Comment by keshav234156** [comment]
ok @jywarren having a look at this.

**Comment by jywarren** [comment]
Hi @NitinBhasneria - so after merging #587 I merged #588 which bumped the NPM version. Let's change this one to `v2.1.2` as it contains the switch over to the `plots2` woofmark branch. Then let's release this ASAP. Thanks a ton!

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "publiclab-editor",
-  "version": "2.1.1",
+  "version": "2.1.2",
```
Comment (jywarren) [inline package.json:3]: OK! And once this merges, can you create a new Release with the release notes here? I think we need to add a few more updates from https://github.com/publiclab/PublicLab.Editor/pulse as well to summarize other changes from folks. @publiclab/editor who else has had changes merged? Can we include https://github.com/publiclab/PublicLab.Editor/pull/568 @keshav234156 ?

Comment (NitinBhasneria) [reply package.json:3]: yes, we can include the #568. 
Also, I have updated the list have a look.

**Comment by sagarpreet-chadha** [comment]
Very exciting 🎉 
cc @NitinBhasneria @keshav234156 @shreyaa-sharmaa @Shulammite-Aso , make sure you add your changes here so Nitin can make the release note 💯

**Comment by NitinBhasneria** [comment]
> @NitinBhasneria can you run `npm install` by checking into this branch? There should be changes in `package-lock.json` also I believe.

yea, sure.

**Comment by shreyaa-sharmaa** [comment]
Looks good!! :tada: :tada:

File: package-lock.json [hunk 2]
```diff
@@ -13298,10 +13301,11 @@
       "dev": true
     },
     "woofmark": {
-      "version": "git://github.com/jywarren/woofmark.git#12e2c1d4559268e2de605ad6084d5dce24ea42c6",
+      "version": "git://github.com/jywarren/woofmark.git#39237b798308e6855acde1a938c18a8e4da62fce",
```
Comment (sagarpreet-chadha) [inline package-lock.json:13304]: Only this should had changed, everything else should be same.

Comment (sagarpreet-chadha) [reply package-lock.json:13304]: See this https://stackoverflow.com/questions/44297803/what-is-the-role-of-the-package-lock-json#:~:text=the%20npm%20docs%3A-,package-lock.,regardless%20of%20intermediate%20

**Comment by jywarren** [comment]
Hi all, thanks! Looks like tests failed, could it be due to the woofmark changes?

**Comment by jywarren** [comment]
Huh. The error is: https://travis-ci.com/github/publiclab/PublicLab.Editor/jobs/370789599#L211

I'll restart it but is there any possibility of an issue in `woofmark`?

```
npm ERR! Unexpected string in JSON at position 516656
```

**Comment by jywarren** [comment]
And noting we had forgotten to build `dist` files in `v2.1.1` so we had to release `v2.1.2` here: https://github.com/publiclab/plots2/pull/8286

So this now becomes `v2.1.3` -- thanks!!!

**Comment by Shulammite-Aso** [comment]
Hi all, i think these should be added here too: 
fix bold&emphasized texts loosing formatting after converting to markdown and back to wysiwyg https://github.com/jywarren/woofmark/pull/15
Update dependencies in plots2 branch https://github.com/jywarren/woofmark/pull/17
Add support for class attribute in links https://github.com/publiclab/PublicLab.Editor/pull/547

**Comment by jywarren** [comment]
Resolved a conflict in package-lock.json - can folks confirm that this is correct by testing it in GitPod? Thanks!

**Comment by jywarren** [comment]
Ah, let's resolve this critical main image upload bug first, then this should be good for final approval and publication! https://github.com/publiclab/PublicLab.Editor/issues/593

**Comment by jywarren** [comment]
As well as this formatting issue to try to get the main image module a little more compact - that would be really awesome! #592 -- any volunteers?

**Comment by jywarren** [comment]
Having tested https://github.com/publiclab/PublicLab.Editor/issues/593 in gitpod, it seems to work there. Can anyone else confirm this locally, installed on `plots2`? I think that may mean we are OK to move forward with this release, although I'd love it if someone stepped up to do a little refining in https://github.com/publiclab/PublicLab.Editor/issues/592 first. Thank you!

**Comment by jywarren** [comment]
Thanks @shreyaa-sharmaa for submitting this PR for the button styling!!! https://github.com/publiclab/PublicLab.Editor/pull/594 Awesome, almost there!!

**Comment by jywarren** [comment]
Hi all! With **Compact Design for main Image Module** #594 by @shreyaa-sharmaa, we are ready to publish this!!!! 🎉 🎉 🎉 🎉

**Comment by jywarren** [comment]
I'm going to merge this and publish it! 🎉 🎉 🎉 🎉 🎉

**Comment by jywarren** [comment]
Technically upon review this would be a minor version number bump of `v2.2.0` according to semantic versioning! So, I'm going to bump it up one more time to that, then release.

**Comment by jywarren** [comment]
Hooray!!! 

https://github.com/publiclab/PublicLab.Editor/releases/tag/v2.2.0

https://www.npmjs.com/package/publiclab-editor

**Comment by jywarren** [comment]
And running plots2 tests here! https://github.com/publiclab/plots2/pull/8297

**Pull request merged**